Mathlib/Order/LiminfLimsup.lean

```diff
@@ -1517,3 +1517,98 @@ theorem liminf_finset_inf [ConditionallyCompleteLinearOrder β] [OrderTop β] {f
   limsup_finset_sup (β := βᵒᵈ) h₁ h₂
 
 end MinMax
+
+section frequently_bounded
+
+variable {R S : Type*} {F : Filter R} [LinearOrder R] [LinearOrder S]
+
+/-- For nontrivial filters in linear orders, coboundedness for `≤` implies frequently boundedness
+from below. -/
+lemma IsCobounded.frequently_ge [NeBot F] (cobdd : IsCobounded (· ≤ ·) F) :
+    ∃ l, ∃ᶠ x in F, l ≤ x := by
+  obtain ⟨t, ht⟩ := cobdd
+  by_cases tbot : IsBot t
+  · refine ⟨t, frequently_of_forall fun r ↦ tbot r⟩
+  obtain ⟨t', ht'⟩ : ∃ t', t' < t := by
+    by_contra!
+    exact tbot this
+  refine ⟨t', ?_⟩
+  intro ev
+  specialize ht t' (by filter_upwards [ev] with _ h using (not_le.mp h).le)
+  apply lt_irrefl t' <| lt_of_lt_of_le ht' ht
+
+/-- For nontrivial filters in linear orders, coboundedness for `≥` implies frequently boundedness
+from above. -/
+lemma IsCobounded.frequently_le [NeBot F] (cobdd : IsCobounded (· ≥ ·) F) :
+    ∃ u, ∃ᶠ x in F, x ≤ u :=
```

Comment:
Same here

Reply:
I now at least temporarily added `IsCobounded.eventually_atTop_frequently_le`, although I am not sure of its usefulness, so I would be happy to remove it.